paper/paper/src/main/java/dev/triumphteam/gui/paper/container/type/HopperContainerType.java

```diff
@@ -0,0 +1,47 @@
+package dev.triumphteam.gui.paper.container.type;
+
+import dev.triumphteam.gui.container.type.GuiContainerType;
+import dev.triumphteam.gui.exception.TriumphGuiException;
+import dev.triumphteam.gui.slot.Slot;
+import net.kyori.adventure.text.Component;
+import org.bukkit.Bukkit;
+import org.bukkit.event.inventory.InventoryType;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.InventoryHolder;
+import org.jetbrains.annotations.NotNull;
+
+public class HopperContainerType implements PaperContainerType {
+
+    private static final int LOWER_LIMIT = 0;
+    private static final int UPPER_LIMIT = 5;
+
+    public static @NotNull GuiContainerType of() {
+        return new HopperContainerType();
+    }
+
+    @Override
+    public int mapSlot(final @NotNull Slot slot) {
+        final var realSlot = (slot.column() + (slot.row() - 1) * 9) - 1;
```

Comment:
This can be a bit simplified, and since it's not a chest the 9 doesn't make much sense here. 
```suggestion
        final var realSlot = (slot.column() * slot.row()) - 1;
```